server/service/src/cold_chain/query_temperature_breach.rs

```diff
@@ -46,6 +46,34 @@ pub fn get_temperature_breach(
     }
 }
 
+pub fn get_max_or_min_log_temperature(
+    breach: &TemperatureBreachRow,
+    logs: &Vec<TemperatureLog>,
+) -> Option<f64> {
+    let mut min_or_max_option = None;
+    let mut min_or_max = 0.0;
+
+    for log in logs {
+        if min_or_max_option.is_none() {
+            min_or_max = log.temperature_log_row.temperature;
+        }
+
+        if log.temperature_log_row.temperature < breach.threshold_minimum
+            && log.temperature_log_row.temperature < min_or_max
+        {
+            min_or_max = log.temperature_log_row.temperature;
+            min_or_max_option = Some(min_or_max);
+        } else if log.temperature_log_row.temperature > breach.threshold_maximum
+            && log.temperature_log_row.temperature > min_or_max
+        {
+            min_or_max = log.temperature_log_row.temperature;
+            min_or_max_option = Some(min_or_max);
```

Comment:
Forgive my lack of knowledge in this area... is an excursion breach only in one direction?

I.e. could this list of logs include some logs below min and some above max? In which case you'd need to consider which was further out of range to be the min/max??

Reply:
I think the intention is that there should be two excursion created if we somehow went from below min the above max without anything in between.

Right now I don't think excursion are really used, so I think this solution of the last min or max is good enough for 99% of situations.